Final part of #230: clarity + the flow of information (#265)
UCL/TDMS: README.md

# Time-Domain Maxwell Solver

TDMS, the Time Domain Maxwell Solver, is a hybrid C++ and MATLAB tool for simulating light propagation through a medium by solving Maxwell's equations.
For further details about the method, please refer to the [PDF documentation](https://github.com/UCL/TDMS/blob/gh-doc/masterdoc.pdf).

![The normed z-component of the H field incident on a cylinder](doc/assets/HzNormBanner.png)

## Getting started

> To use TDMS, it needs to be built against [FFTW](https://www.fftw.org/) and [MATLAB](https://www.mathworks.com/products/matlab.html), which must be downloaded and installed before compiling TDMS.

Author: ```suggestion
To use TDMS, it needs to be built against [FFTW](https://www.fftw.org/) and [MATLAB](https://www.mathworks.com/products/matlab.html), which must be downloaded and installed first.
```

> To use TDMS, it needs to be built against [FFTW](https://www.fftw.org/) and [MATLAB](https://www.mathworks.com/products/matlab.html), which must be downloaded and installed before compiling TDMS.

Author: How about "first"?

$ make install
```
If CMake cannot find MATLAB, FFTW, or install to the default installat prefix, uncomment the relevant line(s) and modify the path(s) accordingly.

<details>
<summary>Mac-specific instructions</summary>

To compile TDMS on a Mac, you will need an x86 compiler with libraries for OpenMP.
You can install these using [Homebrew](https://brew.sh) with the command:

```{sh}
brew install llvm
```

After installing with Homebrew, you may need to set the following CMake arguments:

```{sh}
-DCMAKE_CXX_COMPILER=/Users/username/.local/homebrew/opt/llvm/bin/clang++
-DOMP_ROOT=/Users/username/.local/homebrew/opt/llvm/
-DCXX_ROOT=/Users/username/.local/homebrew/opt/llvm
```

On an ARM Mac, you will need to install the x86 version of Homebrew.
To do so, use the following commands:

```{sh}
arch -x86_64 zsh
arch -x86_64 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
arch -x86_64 /usr/local/bin/brew install llvm
```
</details>

You can check that `tdms` was installed correctly and is in your `PATH` by running:
```{sh}
tdms --help
```
in a new terminal.

## How to run

You can run TDMS either directly or from a MATLAB script.

> For beginners, we recommend starting with the demonstration MATLAB script. which you can find in the `examples/arc_01` directory.

Author: ```suggestion
For beginners, we recommend starting with the demonstration MATLAB script, which you can find in the `examples/arc_01` directory.
```

Move into this directory, launch MATLAB, and run the MATLAB script [`run_pstd_bscan.m`](https://github.com/UCL/TDMS/blob/main/examples/arc_01/run_pstd_bscan.m).
This script will generate the input to TDMS, run TDMS, and display sample output.

If you want to run TDMS standalone at the command line, the basic operation is with two arguments: an input file containing simulation parameters, and an output file name.
You can choose between two solver methods: pseudo-spectral time-domain (PSTD, the default) or finite-difference (FDTD, with option `--finite-difference`).

TDMS is parallelized with [OpenMP](https://en.wikipedia.org/wiki/OpenMP).
You can set the maximum number of threads using the `OMP_NUM_THREADS` environment variable before calling the TDMS executable.

```{sh}
export OMP_NUM_THREADS=4 # for example
```

## Citation

> If you use TDMS in your research and found it helpful, please cite this work.

Author: ```suggestion
If you use TDMS in your research and find it helpful, please cite this work.
```

Author: I can't even blame ChatGPT for that. It was me ruining the tenses.
